File: lead-validation-international-dot-net/REST/LVIResponse.cs
using System.Runtime.Serialization;
using System.Linq;

namespace lead_validation_international_dot_net.REST
{
    /// <summary>
    /// Response object for the Lead Validation International REST API, containing information components and phone contact details.
    /// </summary>
    [DataContract]
    public class LVIResponse
    {
        public string OverallCertainty { get; set; }
        public string OverallQuality { get; set; }
        public string LeadType { get; set; }
        public string LeadCountry { get; set; }
        public string NoteCodes { get; set; }
        public string NoteDesc { get; set; }
        public string NameCertainty { get; set; }
        public string NameQuality { get; set; }
        public string FirstNameLatin { get; set; }
        public string LastNameLatin { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string NameNoteCodes { get; set; }
        public string NameNoteDesc { get; set; }
        public string AddressCertainty { get; set; }
        public string AddressQuality { get; set; }
        public string AddressResolutionLevel { get; set; }
        public string AddressLine1 { get; set; }
        public string AddressLine2 { get; set; }
        public string AddressLine3 { get; set; }
        public string AddressLine4 { get; set; }
        public string AddressLine5 { get; set; }
        public string AddressLocality { get; set; }
        public string AddressAdminArea { get; set; }
        public string AddressPostalCode { get; set; }
        public string AddressCountry { get; set; }
        public string AddressNoteCodes { get; set; }
        public string AddressNoteDesc { get; set; }
        public string EmailCertainty { get; set; }
        public string EmailQuality { get; set; }
        public string EmailCorrected { get; set; }
        public string EmailNoteCodes { get; set; }
        public string EmailNoteDesc { get; set; }
        public string IPCertainty { get; set; }
        public string IPQuality { get; set; }
        public string IPLocality { get; set; }
        public string IPAdminArea { get; set; }
        public string IPCountry { get; set; }
        public string IPNoteCodes { get; set; }
        public string IPNoteDesc { get; set; }
        public string Phone1Certainty { get; set; }
        public string Phone1Quality { get; set; }
        public string Phone1Locality { get; set; }
        public string Phone1AdminArea { get; set; }
        public string Phone1Country { get; set; }
        public string Phone1NoteCodes { get; set; }
        public string Phone1NoteDesc { get; set; }
        public string Phone2Certainty { get; set; }
        public string Phone2Quality { get; set; }
        public string Phone2Locality { get; set; }
        public string Phone2AdminArea { get; set; }
        public string Phone2Country { get; set; }
        public string Phone2NoteCodes { get; set; }
        public string Phone2NoteDesc { get; set; }
        public string BusinessCertainty { get; set; }
        public string BusinessQuality { get; set; }
        public string BusinessName { get; set; }
        public string BusinessDomain { get; set; }
        public string BusinessEmail { get; set; }
        public string BusinessNoteCodes { get; set; }
        public string BusinessNoteDesc { get; set; }
        public InformationComponent[] InformationComponents { get; set; }
        public PhoneContact PhoneContact { get; set; }
        public Error Error { get; set; }

        public override string ToString()
        {
            string informationComponents = InformationComponents?.Any() == true
                ? string.Join(", ", InformationComponents.Select(c => c.ToString()))
                : "None";
            string phoneContact = PhoneContact != null ? PhoneContact.ToString() : "None";
            string error = Error != null ? Error.ToString() : "None";

            return $"LVI Response: OverallCertainty = {OverallCertainty}, OverallQuality = {OverallQuality}, " +
                   $"LeadType = {LeadType}, LeadCountry = {LeadCountry}, NoteCodes = {NoteCodes}, NoteDesc = {NoteDesc}, " +
                   $"NameCertainty = {NameCertainty}, NameQuality = {NameQuality}, FirstNameLatin = {FirstNameLatin}, " +
                   $"LastNameLatin = {LastNameLatin}, FirstName = {FirstName}, LastName = {LastName}, " +
                   $"NameNoteCodes = {NameNoteCodes}, NameNoteDesc = {NameNoteDesc}, AddressCertainty = {AddressCertainty}, " +
                   $"AddressQuality = {AddressQuality}, AddressResolutionLevel = {AddressResolutionLevel}, " +
                   $"AddressLine1 = {AddressLine1}, AddressLine2 = {AddressLine2}, AddressLine3 = {AddressLine3}, " +
                   $"AddressLine4 = {AddressLine4}, AddressLine5 = {AddressLine5}, AddressLocality = {AddressLocality}, " +
                   $"AddressAdminArea = {AddressAdminArea}, AddressPostalCode = {AddressPostalCode}, " +
                   $"AddressCountry = {AddressCountry}, AddressNoteCodes = {AddressNoteCodes}, AddressNoteDesc = {AddressNoteDesc}, " +
                   $"EmailCertainty = {EmailCertainty}, EmailQuality = {EmailQuality}, EmailCorrected = {EmailCorrected}, " +
                   $"EmailNoteCodes = {EmailNoteCodes}, EmailNoteDesc = {EmailNoteDesc}, IPCertainty = {IPCertainty}, " +
                   $"IPQuality = {IPQuality}, IPLocality = {IPLocality}, IPAdminArea = {IPAdminArea}, IPCountry = {IPCountry}, " +
                   $"IPNoteCodes = {IPNoteCodes}, IPNoteDesc = {IPNoteDesc}, Phone1Certainty = {Phone1Certainty}, " +
                   $"Phone1Quality = {Phone1Quality}, Phone1Locality = {Phone1Locality}, Phone1AdminArea = {Phone1AdminArea}, " +
                   $"Phone1Country = {Phone1Country}, Phone1NoteCodes = {Phone1NoteCodes}, Phone1NoteDesc = {Phone1NoteDesc}, " +
                   $"Phone2Certainty = {Phone2Certainty}, Phone2Quality = {Phone2Quality}, Phone2Locality = {Phone2Locality}, " +
                   $"Phone2AdminArea = {Phone2AdminArea}, Phone2Country = {Phone2Country}, Phone2NoteCodes = {Phone2NoteCodes}, " +
                   $"Phone2NoteDesc = {Phone2NoteDesc}, BusinessCertainty = {BusinessCertainty}, BusinessQuality = {BusinessQuality}, " +
                   $"BusinessName = {BusinessName}, BusinessDomain = {BusinessDomain}, BusinessEmail = {BusinessEmail}, " +
                   $"BusinessNoteCodes = {BusinessNoteCodes}, BusinessNoteDesc = {BusinessNoteDesc}, " +
                   $"InformationComponents = [{informationComponents}], PhoneContact = {phoneContact}, Error = {error}";
        }
    }

    /// <summary>
    /// Represents an information component in the Lead Validation International REST API response.
    /// </summary>
    [DataContract]
    public class InformationComponent
    {
        public string Name { get; set; }
        public string Value { get; set; }

        public override string ToString()
        {
            return $"InformationComponent: Name = {Name}, Value = {Value}";
        }
    }

    /// <summary>
    /// Represents phone contact information in the Lead Validation International REST API response.
    /// </summary>
    [DataContract]
    public class PhoneContact
    {
        public string Name { get; set; }
        public string Address { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string Zip { get; set; }
        public string Type { get; set; }

        public override string ToString()
        {
            return $"PhoneContact: Name = {Name}, Address = {Address}, City = {City}, State = {State}, Zip = {Zip}, Type = {Type}";
        }
    }

    /// <summary>
    /// Represents error information in the Lead Validation International REST API response.
    /// </summary>
    [DataContract]
    public class Error
    {
        public string Type { get; set; }
        public string TypeCode { get; set; }
        public string Desc { get; set; }
        public string DescCode { get; set; }
        public string Number { get; set; }

        public override string ToString()
        {
            return $"Error: Type = {Type}, TypeCode = {TypeCode}, Desc = {Desc}, DescCode = {DescCode}, Number = {Number}";
        }
    }
}

File: lead-validation-international-nodejs/src/REST/lvi_response.js
export class InformationComponent {
    constructor(data = {}) {
        this.Name = data.Name;
        this.Value = data.Value;
    }

    toString() {
        return `InformationComponent: Name = ${this.Name}, Value = ${this.Value}`;
    }
}

export class PhoneContact {
    constructor(data = {}) {
        this.Name = data.Name;
        this.Address = data.Address;
        this.City = data.City;
        this.State = data.State;
        this.Zip = data.Zip;
        this.Type = data.Type;
    }

    toString() {
        return `PhoneContact: Name = ${this.Name}, Address = ${this.Address}, City = ${this.City}, State = ${this.State}, Zip = ${this.Zip}, Type = ${this.Type}`;
    }
}

export class ErrorModel {
    constructor(data = {}) {
        this.Type = data.Type;
        this.TypeCode = data.TypeCode;
        this.Desc = data.Desc;
        this.DescCode = data.DescCode;
        this.Number = data.Number;
    }

    toString() {
        return `Error: Type = ${this.Type}, TypeCode = ${this.TypeCode}, Desc = ${this.Desc}, DescCode = ${this.DescCode}`;
    }
}

export class LVIResponse {
    constructor(data = {}) {
        this.OverallCertainty = data.OverallCertainty;
        this.OverallQuality = data.OverallQuality;
        this.LeadType = data.LeadType;
        this.LeadCountry = data.LeadCountry;
        this.NoteCodes = data.NoteCodes;
        this.NoteDesc = data.NoteDesc;
        this.NameCertainty = data.NameCertainty;
        this.NameQuality = data.NameQuality;
        this.FirstNameLatin = data.FirstNameLatin;
        this.LastNameLatin = data.LastNameLatin;
        this.FirstName = data.FirstName;
        this.LastName = data.LastName;
        this.NameNoteCodes = data.NameNoteCodes;
        this.NameNoteDesc = data.NameNoteDesc;
        this.AddressCertainty = data.AddressCertainty;
        this.AddressQuality = data.AddressQuality;
        this.AddressResolutionLevel = data.AddressResolutionLevel;
        this.AddressLine1 = data.AddressLine1;
        this.AddressLine2 = data.AddressLine2;
        this.AddressLine3 = data.AddressLine3;
        this.AddressLine4 = data.AddressLine4;
        this.AddressLine5 = data.AddressLine5;
        this.AddressLocality = data.AddressLocality;
        this.AddressAdminArea = data.AddressAdminArea;
        this.AddressPostalCode = data.AddressPostalCode;
        this.AddressCountry = data.AddressCountry;
        this.AddressNoteCodes = data.AddressNoteCodes;
        this.AddressNoteDesc = data.AddressNoteDesc;
        this.EmailCertainty = data.EmailCertainty;
        this.EmailQuality = data.EmailQuality;
        this.EmailCorrected = data.EmailCorrected;
        this.EmailNoteCodes = data.EmailNoteCodes;
        this.EmailNoteDesc = data.EmailNoteDesc;
        this.IPCertainty = data.IPCertainty;
        this.IPQuality = data.IPQuality;
        this.IPLocality = data.IPLocality;
        this.IPAdminArea = data.IPAdminArea;
        this.IPCountry = data.IPCountry;
        this.IPNoteCodes = data.IPNoteCodes;
        this.IPNoteDesc = data.IPNoteDesc;
        this.Phone1Certainty = data.Phone1Certainty;
        this.Phone1Quality = data.Phone1Quality;
        this.Phone1Locality = data.Phone1Locality;
        this.Phone1AdminArea = data.Phone1AdminArea;
        this.Phone1Country = data.Phone1Country;
        this.Phone1NoteCodes = data.Phone1NoteCodes;
        this.Phone1NoteDesc = data.Phone1NoteDesc;
        this.Phone2Certainty = data.Phone2Certainty;
        this.Phone2Quality = data.Phone2Quality;
        this.Phone2Locality = data.Phone2Locality;
        this.Phone2AdminArea = data.Phone2AdminArea;
        this.Phone2Country = data.Phone2Country;
        this.Phone2NoteCodes = data.Phone2NoteCodes;
        this.Phone2NoteDesc = data.Phone2NoteDesc;
        this.BusinessCertainty = data.BusinessCertainty;
        this.BusinessQuality = data.BusinessQuality;
        this.BusinessName = data.BusinessName;
        this.BusinessDomain = data.BusinessDomain;
        this.BusinessEmail = data.BusinessEmail;
        this.BusinessNoteCodes = data.BusinessNoteCodes;
        this.BusinessNoteDesc = data.BusinessNoteDesc;
        this.InformationComponents = (data.InformationComponents || []).map(ic => new InformationComponent(ic));
        this.PhoneContact = data.PhoneContact ? new PhoneContact(data.PhoneContact) : null;
        this.Error = data.Error ? new ErrorModel(data.Error) : null;
    }

    toString() {
        const infoComponents = this.InformationComponents.length
            ? this.InformationComponents.map(ic => ic.toString()).join(', ')
            : 'None';

        const phoneContact = this.PhoneContact ? this.PhoneContact.toString() : 'None';
        const error = this.Error ? this.Error.toString() : 'None';

        return `LVI Response: OverallCertainty = ${this.OverallCertainty}, OverallQuality = ${this.OverallQuality}, ` +
            `LeadType = ${this.LeadType}, LeadCountry = ${this.LeadCountry}, NoteCodes = ${this.NoteCodes}, NoteDesc = ${this.NoteDesc}, ` +
            `NameCertainty = ${this.NameCertainty}, NameQuality = ${this.NameQuality}, FirstNameLatin = ${this.FirstNameLatin}, ` +
            `LastNameLatin = ${this.LastNameLatin}, FirstName = ${this.FirstName}, LastName = ${this.LastName}, ` +
            `NameNoteCodes = ${this.NameNoteCodes}, NameNoteDesc = ${this.NameNoteDesc}, AddressCertainty = ${this.AddressCertainty}, ` +
            `AddressQuality = ${this.AddressQuality}, AddressResolutionLevel = ${this.AddressResolutionLevel}, ` +
            `AddressLine1 = ${this.AddressLine1}, AddressLine2 = ${this.AddressLine2}, AddressLine3 = ${this.AddressLine3}, ` +
            `AddressLine4 = ${this.AddressLine4}, AddressLine5 = ${this.AddressLine5}, AddressLocality = ${this.AddressLocality}, ` +
            `AddressAdminArea = ${this.AddressAdminArea}, AddressPostalCode = ${this.AddressPostalCode}, AddressCountry = ${this.AddressCountry}, ` +
            `AddressNoteCodes = ${this.AddressNoteCodes}, AddressNoteDesc = ${this.AddressNoteDesc}, EmailCertainty = ${this.EmailCertainty}, ` +
            `EmailQuality = ${this.EmailQuality}, EmailCorrected = ${this.EmailCorrected}, EmailNoteCodes = ${this.EmailNoteCodes}, ` +
            `EmailNoteDesc = ${this.EmailNoteDesc}, IPCertainty = ${this.IPCertainty}, IPQuality = ${this.IPQuality}, ` +
            `IPLocality = ${this.IPLocality}, IPAdminArea = ${this.IPAdminArea}, IPCountry = ${this.IPCountry}, IPNoteCodes = ${this.IPNoteCodes}, ` +
            `IPNoteDesc = ${this.IPNoteDesc}, Phone1Certainty = ${this.Phone1Certainty}, Phone1Quality = ${this.Phone1Quality}, ` +
            `Phone1Locality = ${this.Phone1Locality}, Phone1AdminArea = ${this.Phone1AdminArea}, Phone1Country = ${this.Phone1Country}, ` +
            `Phone1NoteCodes = ${this.Phone1NoteCodes}, Phone1NoteDesc = ${this.Phone1NoteDesc}, Phone2Certainty = ${this.Phone2Certainty}, ` +
            `Phone2Quality = ${this.Phone2Quality}, Phone2Locality = ${this.Phone2Locality}, Phone2AdminArea = ${this.Phone2AdminArea}, ` +
            `Phone2Country = ${this.Phone2Country}, Phone2NoteCodes = ${this.Phone2NoteCodes}, Phone2NoteDesc = ${this.Phone2NoteDesc}, ` +
            `BusinessCertainty = ${this.BusinessCertainty}, BusinessQuality = ${this.BusinessQuality}, BusinessName = ${this.BusinessName}, ` +
            `BusinessDomain = ${this.BusinessDomain}, BusinessEmail = ${this.BusinessEmail}, BusinessNoteCodes = ${this.BusinessNoteCodes}, ` +
            `BusinessNoteDesc = ${this.BusinessNoteDesc}, InformationComponents = [${infoComponents}], PhoneContact = ${phoneContact}, Error = ${error}`;
    }
}

export default LVIResponse